presto-main/src/main/java/com/facebook/presto/Session.java

```diff
@@ -348,6 +350,10 @@ public ClientSession toClientSession(URI server, boolean debug, Duration clientR
                 properties.put(catalog + "." + entry.getKey(), entry.getValue());
             }
         }
+        ImmutableMap.Builder<String, String> encodedStatements = ImmutableMap.builder();
+        for (Entry<String, String> entry : preparedStatements.entrySet()) {
+           encodedStatements.put(urlEncode(entry.getKey()), urlEncode(entry.getValue()));
```

Comment:
The encoding should be done in `StatementClient`, since the encoding is part of the wire protocol.